core/src/main/java/org/mule/runtime/core/routing/MessageFilter.java

```diff
@@ -124,6 +124,7 @@ public void setMuleContext(MuleContext context) {
 
   @Override
   public void setFlowConstruct(FlowConstruct flowConstruct) {
+    super.setFlowConstruct(flowConstruct);
     if (!onUnacceptedFlowConstruct && unacceptedMessageProcessor instanceof FlowConstructAware) {
       ((FlowConstructAware) unacceptedMessageProcessor).setFlowConstruct(flowConstruct);
```

Comment:
I thought we could make filter a MessageProessorOwner, but we can't because intercepting MP's aren't supported.